src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingGraph.java

```diff
@@ -992,44 +982,30 @@ private boolean extendDanglingPathAgainstReference(final DanglingChainMergeHelpe
      * among all sequences added to the current graph.  Will always return a result for maxKmerSize if
      * all smaller kmers had non-unique kmers.
      *
-     * @param minKmerSize the minimum kmer size to consider when constructing the graph
-     * @param maxKmerSize the maximum kmer size to consider
+     * @param kmerSize the kmer size to check for non-unique kmers of
      * @return a non-null NonUniqueResult
      */
-    private NonUniqueResult determineKmerSizeAndNonUniques(final int minKmerSize, final int maxKmerSize) {
+    private Set<Kmer> determineNonUniques(final int kmerSize) {
         final Collection<SequenceForKmers> withNonUniques = getAllPendingSequences();
         final Set<Kmer> nonUniqueKmers = new HashSet<>();
 
-        // go through the sequences and determine which kmers aren't unique within each read
-        for (int kmerSize = minKmerSize ; kmerSize <= maxKmerSize; kmerSize++) {
-            // clear out set of non-unique kmers
-            nonUniqueKmers.clear();
-
-            // loop over all sequences that have non-unique kmers in them from the previous iterator
-            final Iterator<SequenceForKmers> it = withNonUniques.iterator();
-            while ( it.hasNext() ) {
-                final SequenceForKmers sequenceForKmers = it.next();
-
-                // determine the non-unique kmers for this sequence
-                final Collection<Kmer> nonUniquesFromSeq = determineNonUniqueKmers(sequenceForKmers, kmerSize);
-                if ( nonUniquesFromSeq.isEmpty() ) {
-                    // remove this sequence from future consideration
-                    it.remove();
-                } else {
-                    // keep track of the non-uniques for this kmerSize, and keep it in the list of sequences that have non-uniques
-                    nonUniqueKmers.addAll(nonUniquesFromSeq);
-                }
-            }
+        // loop over all sequences that have non-unique kmers in them from the previous iterator
```

Comment:
Is this comment still necessary now that you're not looping over multiple kmer sizes?